sample: syst: Add config for testing on intel_adsp_cavs platforms

aasthagr opened the pull request:
Adding configurations for enabling this test on cavs platforms.
This patch also contains a fix to mipi_syst library which resolves
the memory alignment issue across different architectures.

Fixes #43344

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

----------------
This is the old change set.

See PR #44491 for the most recent change set.

aasthagr committed:
sample: syst: Add config for testing on intel_adsp_cavs platforms

Adding configurations for enabling this test on cavs platforms.
This patch also contains a fix to mipi_syst library which resolves
the memory alignment issue across different architectures.

Fixes #43344

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

yerabolu reviewed (commented):
Sorry, forgot to run the west update to pull in the mipi-sys-t changes. After doing the west update, sample.logger.syst.adsp_cavs sample passed.

yerabolu commented on samples/subsys/logging/syst/sample.yaml:
```diff
@@ -10,16 +10,19 @@ tests:
     tags: logging
     harness: console
     harness_config:
       type: one_line
```
should this be a multi_line harness_config type?

nordic-krch commented on west.yml:
```diff
@@ -179,7 +179,7 @@ manifest:
       groups:
         - debug
-      revision: d9da086b11cda494d85f4d8a9829f505c2d5e380
+      revision: pull/9/head
```
pr got merged

nashif committed:
manifest: update sys-t revision to a5163c180

Update sys-t module revision to a5163c180.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented on samples/subsys/logging/syst/sample.yaml:
```diff
@@ -28,14 +31,26 @@ tests:
+        - "SYST Sample Execution Completed"
+  sample.logger.syst.adsp_cavs:
+    extra_args: OVERLAY_CONFIG=overlay_cavs.conf
+    platform_allow: intel_adsp_cavs15 intel_adsp_cavs18 intel_adsp_cavs20 intel_adsp_cavs25
```
doing this you are missing all the other tests on the listed platforms. You should add the overlay in boards/<board name> instead and remove all platform_allow entries here and instead using integration_platforms to allow more coverage.

peter-mitsis committed:
samples: syst: Update sample.yaml

Updates the sample.yaml file to use 'integration_platforms' instead
of 'platform_allow'. Related to this, the overlay '.conf' files now
reside under the 'boards' sub-directory.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>
